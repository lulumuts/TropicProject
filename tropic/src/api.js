const axios = require("axios");
async function getAllData() {
  const response = await axios.get(
    "https://jsonplaceholder.typicode.com/todos"
  );
  console.log(response.data);
  return response.data;
}

module.exports = getAllData;
