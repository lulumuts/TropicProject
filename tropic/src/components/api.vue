<template>
  <div>
    <h1>{{ msg }}</h1>
    <input v-model="search" />
    <div v-for="item in searchFilter" :key="item.id">
      {{ item }}
    </div>
  </div>
</template>

<script>
export default {
  name: "api",
  props: { msg: String },
  data() {
    return {
      todos: [],
      search: "",
    };
  },
  computed: {
    searchFilter() {
      return this.todos.filter((x) => {
        return x.title.toLowerCase().includes(this.search.toLowerCase());
      });
    },
  },
  mounted() {
    this.allData();
  },
  methods: {
    async allData() {
      try {
        const response = await this.axios.get(
          "https://jsonplaceholder.typicode.com/todos"
        );
        return (this.todos = response.data);
      } catch (error) {
        console.log(error);
      }
    },
  },
};
</script>
