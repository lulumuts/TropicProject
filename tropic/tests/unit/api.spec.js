import { shallowMount } from "@vue/test-utils";
import api from "@/components/api.vue";
const axios = require("axios");
const getAllData = require("@/api");
jest.mock("axios");

describe("api.vue", () => {
  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = shallowMount(api, {
      props: { msg },
    });
    expect(wrapper.text()).toMatch(msg);
  });

  it("renders snapshot when list is not empty", async () => {
    const todos = {
      data: [
        {
          userId: 1,
          id: 1,
          title: "delectus aut autem",
          completed: false,
        },
        {
          userId: 1,
          id: 2,
          title: "quis ut nam facilis et officia qui",
          completed: false,
        },
      ],
    };
    const wrapper = shallowMount(api, { todos });
    expect(wrapper.element).toMatchSnapshot();
  });
  // with jest uses a mock value to test the api vs making a real request

  it("returns the api data", async () => {
    axios.get.mockResolvedValue({
      data: [
        {
          userId: 1,
          id: 1,
          title: "delectus aut autem",
          completed: false,
        },
        {
          userId: 1,
          id: 2,
          title: "quis ut nam facilis et officia qui",
          completed: false,
        },
      ],
    });

    const title = await getAllData();
    expect(title).toEqual([
      {
        userId: 1,
        id: 1,
        title: "delectus aut autem",
        completed: false,
      },
      {
        userId: 1,
        id: 2,
        title: "quis ut nam facilis et officia qui",
        completed: false,
      },
    ]);
  });
});
