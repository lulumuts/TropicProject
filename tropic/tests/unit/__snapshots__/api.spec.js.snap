// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api.vue renders snapshot when list is not empty 1`] = `
<div>
  <h1 />
  <input />
  
  
</div>
`;

exports[`api.vue returns the details of the first user 1`] = `
<div>
  <h1 />
  <input />
  
  
</div>
`;
